### DISCUSSION TITLE
settings.json is always restored no matter what changes made.

### DISCUSSION THREAD

**Discussion by jianlianggao** [transmission/transmission#4432] [Q&A]
### What is the issue?

Dear all,

I started transmission 5 months ago using Docker engine on Ubuntu 18.04 bionic LTS. Then I realized that the rpc authentication was set to false by default. I have been trying different ways to edit the settings.json file by changing rpc authentication to true, adding username and password. But when I restarted the Docker container, the settings.json has always been restored to the one that wasn't edited. 

I tried: 
(1) stopped container, edited settings.json and saved, started the container.
(2) stopped docker engine, edited settings.json and saved, started docker engine.
(3) entered container, edited settings.json and saved, checked the changes on host system, restarted the container.
(4) stopped docker engine, edited settings.json and saved, changed settings.json to read-only, started docker engine.
All the above methods failed, the settings.json was restored once the container or docker was started (re-started).

Can anyone tell me what has gone wrong please? Can you please also share how to make the change take effect to enable rpc authentication and password? Thank you in advance.

The following message came from docker inspect

   {
        "Id": "534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0",
        "Created": "2022-07-18T11:58:26.426966244Z",
        "Path": "/init",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 5961,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2022-12-20T21:38:52.251633548Z",
            "FinishedAt": "2022-12-20T17:17:41.038133386Z"
        },
        "Image": "sha256:7dcd6a71dec500837358021a6a445a4a4b85b20b615fff7e6e149f7b5a51e8dd",
        "ResolvConfPath": "/var/lib/docker/containers/534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0/hostname",
        "HostsPath": "/var/lib/docker/containers/534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0/hosts",
        "LogPath": "/var/lib/docker/containers/534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0/534993d8859f9fa93427d402029c98a9313be36fd622cdd107ced28703098bf0-json.log",
        "Name": "/transmissionSource",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/srv/transmission/config:/config",
                "/srv/transmission/downloads:/downloads",
                "/srv/transmission/watch:/watch",
                "/mnt/release-pilot:/release-pilot"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "51413/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "51413"
                    }
                ],
                "51413/udp": [
                    {
                        "HostIp": "",
                        "HostPort": "51413"
                    }
                ],
                "9091/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "9091"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/244bd4d59a613f2ccfee2695840f7a7327875b9f55ea21619c321d89ac782903-init/diff:/var/lib/docker/overlay2/d76b5303ce3c5b7f59328f227bbf02dc0d64a532711bb457fdfba8c824ddd4ba/diff:/var/lib/docker/overlay2/fcd48f34bc65ab46de7832ac728f0d5cc4a02489bb8e81cebc8a149c5f63504b/diff:/var/lib/docker/overlay2/a41c4220b90f5f9ff1da0ba77658b88dc59827b03cd63823945f4f862a2a073d/diff:/var/lib/docker/overlay2/fe202040f9c59648c4f250b7d98e3ffe7db113c475cb3f51291ef91a750ee418/diff:/var/lib/docker/overlay2/794b11c6e9c82a77f1a23a82b135aeb18d2be23d6435a54d13f4cf5dbc7f8e35/diff:/var/lib/docker/overlay2/5788f45dc66dc1512f64bedc9682fe9439a7ac43049fa04cb5a2683e3045c536/diff:/var/lib/docker/overlay2/692417643d9ccd74911e28be9016ea2dd1e0ed1e963e34a418a9de08fd49934b/diff:/var/lib/docker/overlay2/caac82afa7e885a4fc439cd80f3d3f2e764d438ad8813edb65290423179c9076/diff",
                "MergedDir": "/var/lib/docker/overlay2/244bd4d59a613f2ccfee2695840f7a7327875b9f55ea21619c321d89ac782903/merged",
                "UpperDir": "/var/lib/docker/overlay2/244bd4d59a613f2ccfee2695840f7a7327875b9f55ea21619c321d89ac782903/diff",
                "WorkDir": "/var/lib/docker/overlay2/244bd4d59a613f2ccfee2695840f7a7327875b9f55ea21619c321d89ac782903/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/srv/transmission/config",
                "Destination": "/config",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/srv/transmission/downloads",
                "Destination": "/downloads",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/mnt/release-pilot",
                "Destination": "/release-pilot",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/srv/transmission/watch",
                "Destination": "/watch",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "534993d8859f",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "51413/tcp": {},
                "51413/udp": {},
                "9091/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PGID=1121",
                "PUID=1642",
                "TZ=Europe/London",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PS1=$(whoami)@$(hostname):$(pwd)\\$ ",
                "HOME=/root",
                "TERM=xterm"
            ],
            "Cmd": null,
            "Image": "linuxserver/transmission",
            "Volumes": {
                "/config": {},
                "/downloads": {},
                "/watch": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/init"
            ],
            "OnBuild": null,
            "Labels": {
                "build_version": "Linuxserver.io version:- 3.00-r2-ls80 Build-date:- 2021-03-04T03:42:29+00:00",
                "maintainer": "aptalca"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "0c62d98f406bebe95f065048c53f9cb44cbb27438666eb698ffd90d904bdb361",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "51413/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "51413"
                    }
                ],
                "51413/udp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "51413"
                    }
                ],
                "9091/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "9091"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/0c62d98f406b",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "4390a0596562351baf5950aaa22b5e20de5a2b32c3327136d4cc1cef7e80e871",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "543441407a4d8febc69720a4c378d04de88efc7e5a9af0be05485dd32301b989",
                    "EndpointID": "4390a0596562351baf5950aaa22b5e20de5a2b32c3327136d4cc1cef7e80e871",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02",
                    "DriverOpts": null
                }
            }
        }
    }

Best,
J

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

Transmission 3.00 (bb6b5a062e)

**Comment by TomHsiung** [comment]
You have to stop the daemon service first. Then you can edit and save the `settings.json` file.
